x-pack/plugins/observability/public/pages/rule_details/index.tsx

```diff
@@ -108,6 +110,7 @@ export function RuleDetailsPage() {
   const [editFlyoutVisible, setEditFlyoutVisible] = useState<boolean>(false);
   const [isRuleEditPopoverOpen, setIsRuleEditPopoverOpen] = useState(false);
   const [esQuery, setEsQuery] = useState<{ bool: BoolQuery }>();
+  const [defaultAlertTimeRange] = useState(getDefaultAlertSummaryTimeRange);
```

Comment:
Do we need to use a state for this? Can it be just a variable in this scope, or a constant in the module?

Reply:
The reason for having this state is to run `getAbsoluteTimeRange` only once during the rendering of this component. `getAbsoluteTimeRange` will return the absolute time for the last 30 days at the time of the execution and I didn't want this value to be created more than one time in this component but I also didn't want to have it created when the module is loaded so I used the state in this case.

Reply:
nit - I also saw that people use `useRef` to do that to avoid the state

Reply:
In this case, I'll keep it in the state to be able to refresh it in case we want to.